File: DataSet/src/main/java/de/kimanufaktur/nsm/semanticDistanceMeasures/data/DataSet.java
/*
 * Copyright (C) Johannes Fähndrich - All Rights Reserved.
 * Unauthorized copying of this file, via any medium is strictly
 * prohibited Proprietary and confidential.
 * Written by Johannes Fähndrich <faehndrich@gmail.com.com>,  2011
 *
 */

package de.kimanufaktur.nsm.semanticDistanceMeasures.data;

import de.kimanufaktur.nsm.semanticDistanceMeasures.DataExample;
import de.kimanufaktur.nsm.semanticDistanceMeasures.SimilarityPair;

import java.util.Collection;
import java.util.List;

/**
 * Created by Johannes Fähndrich on 10.06.18 as part of his dissertation.
 */
public interface DataSet {
    List<DataExample> ReadExampleDataSet();

    Collection<SimilarityPair> Normalize(List<SimilarityPair> list2Normlize);
}

File: semantic_decomposition_experiments/dataset/data_set.py
from abc import ABC, abstractmethod
from typing import Collection, List
from .data_example import DataExample
from .similarity_pair import SimilarityPair


class DataSet(ABC):
    @abstractmethod
    def read_example_data_set(self) -> List[DataExample]: ...

    @abstractmethod
    def normalize(self) -> Collection[SimilarityPair]: ...
